vllm/model_executor/sampling_metadata.py

```diff
@@ -538,6 +548,14 @@ def from_lists(cls, temperatures: List[float], top_ps: List[float],
             extra_seeds_gpu = None
         sampling_seeds_gpu = sampling_seeds_gpu[:num_base_seeds]
 
+        if do_penalties:
+            prompt_tokens_gpu = prompt_tensor.to(device=device, non_blocking=True)
+            output_tokens_gpu = output_tensor.to(device=device, non_blocking=True)
+        else:
+            empty_tensor = torch.empty(0, device=device, dtype=torch.long)
+            prompt_tokens_gpu = empty_tensor
+            output_tokens_gpu = empty_tensor
```

Comment:
Is there any concern with uninitialized values possibly causing errors or NaNs in the kernel?

Reply:
If those two tensors are empty, then it means they will never be used.